Configure dependabot

aw1203 opened the pull request:
**Issue:**
Ties into: https://github.com/uoy-trials/about-dev/issues/5

Configure dependabot to check github actions and nuget packages, open PR's as required.

**Fix:**
Added configuration yml file

**Notes**
This has a property to ignore major version updates so w don't get alerts for packages updated for .NET 9

**Suggested Testing**
Check what alerts are raised!

**Review Tasks**

[PR review wiki](https://github.com/uoy-trials/about-dev/wiki/GitHub-Review-Process)
- [ ] close any linked change requests
- [ ] add relevant labels to the pr
- [ ] create a new release


ChaReq link: None

closes #17

edh519 reviewed (approved):
Tested on a personal repository - all is working fine!
@uoy-trials/code-reviewers this can be merged!

aw1203 commented:
I've added in upy-trials registry and changed nuget updates to allow access to all registries (those defined at top level and all default ie public ones)

edh519 commented:
There is a breaking change from dependabot that is prohibiting this file from running correctly:
https://github.com/dependabot/dependabot-core/issues/11525

edh519 reviewed (approved):
I've tested this on a private repo and it's all working fine!
@tp1045  this can be merged.

note: repo is not on chareq so no message required.
